sdk/Sdk.Generators/FunctionMetadataProviderGenerator/FunctionMetadataProviderGenerator.Parser.cs

```diff
@@ -157,11 +186,63 @@ private bool TryGetMethodOutputBinding(MethodDeclarationSyntax method, SemanticM
                 return true;
             }
 
+            private bool TryGetRetryOptionsFromAtttribute(AttributeData attribute, out GeneratorRetryOptions? retryOptions)
+            {
+                retryOptions = null;
+
+                if (TryGetAttributeProperties(attribute, null, out IDictionary<string, object?>? attrProperties))
+                {
+                    retryOptions = new GeneratorRetryOptions();
+
+                    // Would not expect this to fail since MaxRetryCount is a required value of a retry policy attribute
+                    if (attrProperties!.TryGetValue(Constants.RetryConstants.MaxRetryCountKey, out object? maxRetryCount))
+                    {
+                        retryOptions.MaxRetryCount = maxRetryCount!.ToString();
+                    }
+
+                    // Check which strategy is being used by checking the attribute class
+                    if (SymbolEqualityComparer.Default.Equals(attribute.AttributeClass, _knownFunctionMetadataTypes.FixedDelayRetryAttribute))
+                    {
+                        retryOptions.Strategy = Constants.RetryConstants.FixedDelayRetryName;
```

Comment:
small nit: while this is all internal and is not a big concern. An enum here is more fitting still. Primarily because it is easier and cheaper to compare it later on (enum equality is much cheaper than string equality).

No need to fix this if you aren't bothered, but something to keep in mind for later. Don't be afraid to create multiple internal types to drive your scenario.